core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1872,6 +1872,46 @@ class WorkspaceService(protected val userInfo: UserInfo,
     }
   }
 
+  def adminListWorkspacesWithInvalidEntityCaches: Future[PerRequestMessage] = {
+    asFCAdmin {
+      for {
+        workspacesAndErrors <- dataSource.inTransaction { dataAccess =>
+          for {
+            workspaceIdsAndErrors <- dataAccess.entityCacheQuery.listInvalidCaches
+            workspaces <- dataAccess.workspaceQuery.listByIds(workspaceIdsAndErrors.map(_._1), Option(WorkspaceAttributeSpecs(all=false)))
+          } yield {
+            val errorMap = workspaceIdsAndErrors.toMap
+            workspaces.map { w =>
+              (w, errorMap.get(w.workspaceIdAsUUID).flatten.getOrElse(""))
+            }
+          }
+        }
+        results <- Future.traverse(workspacesAndErrors) { workspaceAndError =>
+          val workspace = workspaceAndError._1
+          loadResourceAuthDomain(SamResourceTypeNames.workspace, workspace.workspaceId, userInfo).map(WorkspaceDetails(workspace, _)).map { details =>
+            WorkspaceDetailsAndCacheError(details, workspaceAndError._2)
+          }
+        }
+      } yield {
+        RequestComplete(StatusCodes.OK, results)
+      }
+    }
+  }
+
+  def adminResetWorkspaceEntityCaches(ids: Seq[String]) = {
```

Comment:
It looks like if one of caches fails to reset, then the whole operation fails?  I'd be in favor of changing the POST to just accepting a single workspace ID at a time.  I think that makes error handling and messaging to the caller cleaner.